org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/XtextGeneratorResourceSetInitializer.java

```diff
@@ -0,0 +1,152 @@
+/**
+ * Copyright (c) 2015, 2020 itemis AG (http://www.itemis.eu) and others.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.xtext.xtext.generator;
+
+import java.util.List;
+
+import org.apache.log4j.Logger;
+import org.eclipse.emf.codegen.ecore.genmodel.GenModel;
+import org.eclipse.emf.codegen.ecore.genmodel.GenModelPackage;
+import org.eclipse.emf.common.util.URI;
+import org.eclipse.emf.common.util.WrappedException;
+import org.eclipse.emf.ecore.EPackage;
+import org.eclipse.emf.ecore.EcorePackage;
+import org.eclipse.emf.ecore.resource.Resource;
+import org.eclipse.emf.ecore.resource.ResourceSet;
+import org.eclipse.emf.mwe.utils.GenModelHelper;
+import org.eclipse.emf.mwe.utils.StandaloneSetup;
+import org.eclipse.xtext.resource.IResourceServiceProvider;
+import org.eclipse.xtext.xbase.lib.Procedures.Procedure1;
+
+import com.google.common.collect.Iterables;
+
+/**
+ * Initializes a resource set in order to load the grammar of a language. The
+ * resources to include are specified via
+ * {@link XtextGeneratorLanguage#addReferencedResource(String)}.
+ */
+public class XtextGeneratorResourceSetInitializer {
+	private static final Logger LOG = Logger.getLogger(XtextGeneratorResourceSetInitializer.class);
+
+	public void initialize(ResourceSet resourceSet, List<String> referencedResources) {
+		StandaloneSetup delegate = new StandaloneSetup();
+		delegate.setResourceSet(resourceSet);
+		resourceSet.getPackageRegistry().put(EcorePackage.eNS_URI, EcorePackage.eINSTANCE);
+		referencedResources.forEach((String it) -> {
+			loadResource(it, resourceSet);
+		});
+		registerGenModels(resourceSet);
+		registerEPackages(resourceSet);
+	}
+
+	private void loadResource(String loadedResource, ResourceSet resourceSet) {
+		URI loadedResourceUri = URI.createURI(loadedResource);
+		ensureResourceCanBeLoaded(loadedResourceUri, resourceSet);
+		resourceSet.getResource(loadedResourceUri, true);
+	}
+
+	private void ensureResourceCanBeLoaded(URI loadedResource, ResourceSet resourceSet) {
+		String fileExtension = loadedResource.fileExtension();
+		if (fileExtension != null) {
+			switch (fileExtension) {
+			case "genmodel":
+				GenModelPackage.eINSTANCE.getEFactoryInstance();
+				IResourceServiceProvider resourceServiceProvider = IResourceServiceProvider.Registry.INSTANCE
+						.getResourceServiceProvider(loadedResource);
+				if (resourceServiceProvider == null) {
+					try {
+						Class<?> genModelSupport = Class.forName("org.eclipse.emf.codegen.ecore.xtext.GenModelSupport");
+						Object instance = genModelSupport.getDeclaredConstructor().newInstance();
+						genModelSupport.getDeclaredMethod("createInjectorAndDoEMFRegistration").invoke(instance);
+					} catch (ClassNotFoundException e) {
+						LOG.debug(
```

Comment:
1) For CNF, an error message is logged on lines 83, and 98. Here, a debug message is logged
2) Just to be consistent with other log messages, you might like to change it to - Couldn't initialize Ecore support. It is possible that the library "org.eclipse.xtext.ecore" is missing from the project classpath.